test/test_node.py

```diff
@@ -0,0 +1,121 @@
+"""Unit tests for the RemoteNode and LocalNode classes."""
+import unittest
+
+import canopen
+
+from .util import SAMPLE_EDS
+
+
+def count_subscribers(network: canopen.Network) -> int:
+    """Count the number of subscribers in the network."""
+    return sum(
+        len(n) for n in network.subscribers.values()
+    )
+
+
+class TestLocalNode(unittest.TestCase):
+    """
+    Test local node.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.network = canopen.Network()
+        cls.network.NOTIFIER_SHUTDOWN_TIMEOUT = 0.0
+        cls.network.connect("test", interface="virtual")
+
+        cls.node = canopen.LocalNode(2, SAMPLE_EDS)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.network.disconnect()
+
+    def test_associate_network(self):
+
+        # Need to store the number of subscribers before associating because
+        # the current network implementation automatically adds subscribers
+        # to the list
```

Comment:
```suggestion
        # Need to store the number of subscribers before associating because the
        # network implementation automatically adds subscribers to the list
```